tensorflow_probability/python/positive_semidefinite_kernels/polynomial_test.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2018 The TensorFlow Probability Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ============================================================================
+"""Tests for Polynomial."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+
+import numpy as np
+import tensorflow as tf
+
+from tensorflow_probability import positive_semidefinite_kernels as psd_kernels
+
+
+class PolynomialTest(tf.test.TestCase):
+
+  def testMismatchedFloatTypesAreBad(self):
+    # Should be OK (float32 fallback).
+    self.polynomial = psd_kernels.Polynomial(bias_variance=0, slope_variance=1, exponent=1)
+    # Should be OK.
+    psd_kernels.Polynomial(
+      bias_variance=np.float32(1.),
+      slope_variance=1.,
+      exponent=1.)
+    with self.assertRaises(TypeError):
+      psd_kernels.Polynomial(
+        bias_variance=np.float32(1.),
+        slope_variance=np.float64(1.),
+        exponent=1.
+      )
+
+  def testValidateArgs(self):
+    k = psd_kernels.Polynomial(
+      bias_variance=-1.,
+      slope_variance=-1.,
+      exponent=-1.,
+      validate_args=True)
+    with self.assertRaises(tf.errors.InvalidArgumentError):
+      self.evaluate(k.bias_variance)
+    with self.assertRaises(tf.errors.InvalidArgumentError):
+      self.evaluate(k.slope_variance)
+    with self.assertRaises(tf.errors.InvalidArgumentError):
+      self.evaluate(k.exponent)
+    # But `None`'s are ok
+    k = psd_kernels.Polynomial(
+      bias_variance=None,
+      slope_variance=None,
+      exponent=None,
+      validate_args=True)
+    self.evaluate(k.apply([[1.]], [[1.]]))
+
+  def testNoneShapes(self):
+    k = psd_kernels.Polynomial(
+      bias_variance=np.reshape(np.arange(12.), [2, 3, 2]))
+    self.assertEqual([2, 3, 2], k.batch_shape.as_list())
+
+  def testFloat32(self):
+    k = psd_kernels.Polynomial(
+      bias_variance=0.,
+      slope_variance=1.,
+      exponent=1.,
+      feature_ndims=1)
+    x = np.ones([5, 3], np.float32)
+    y = np.ones([5, 3], np.float32)
+    k.apply(x, y)
+
+  def testFloat64(self):
+    k = psd_kernels.Polynomial(
+      bias_variance=np.float64(0.),
+      slope_variance=np.float64(1.),
+      exponent=np.float64(1.),
+      feature_ndims=1)
+    x = np.ones([5, 3], np.float64)
+    y = np.ones([5, 3], np.float64)
+    k.apply(x, y)
+
+  def testShapesAreCorrect(self):
```

Comment:
Don't have a strong preference here, but I might opt for something like using @parameterized.named_parameters:
https://github.com/tensorflow/probability/blob/master/tensorflow_probability/python/bijectors/batch_normalization_test.py#L172

That way you can specify the cases outside this block, and reuse the same code (but it might be more difficult in this case).